src/KM_Controls.pas

```diff
@@ -3307,54 +3369,47 @@ function TKMNumericEdit.KeyEventHandled(Key: Word; Shift: TShiftState): Boolean;
 
   //Ctrl can be used as an escape character, e.g. CTRL+B places beacon while chat is open
   if ssCtrl in Shift then
-    Result := (Key in [Ord('C'), Ord('X'), Ord('V')]);
+    Result := (Key in [Ord('A'), Ord('C'), Ord('X'), Ord('V')]);
 end;
 
 
-procedure TKMNumericEdit.KeyPress(Key: Char);
-begin
-  if Length(fText) >= MaxLength then Exit;
-
-  Insert(Key, fText, CursorPos + 1);
-  CursorPos := CursorPos + 1; //Before ValidateText so it moves the cursor back if the new char was invalid
-  ValidateText;
-end;
-
-
-function TKMNumericEdit.KeyUp(Key: Word; Shift: TShiftState): Boolean;
-begin
-  Result := KeyEventHandled(Key, Shift);
-  if inherited KeyUp(Key, Shift) then Exit;
-end;
-
-
-function TKMNumericEdit.MaxLength: Byte;
+function TKMNumericEdit.GetMaxLength: Word;
 begin
   if Max(Abs(ValueMax), Abs(ValueMin)) <> 0 then
-    Result := Trunc(Log10(Max(Abs(ValueMax), Abs(ValueMin)))) + 1
+    Result := Trunc(Max(Log10(Abs(ValueMax)) + Byte(ValueMax < 0), Log10(Abs(ValueMin)) + Byte(ValueMin < 0))) + 1

```

Comment:
fix MaxLength calculation - we allow negative number, so must rely also on negative numbers (and minus sign too)